engine/context/src/main/java/io/deephaven/engine/context/ExecutionContext.java

```diff
@@ -112,7 +116,22 @@ public ExecutionContext withSystemic(boolean isSystemic) {
         if (isSystemic == this.isSystemic) {
             return this;
         }
-        return new ExecutionContext(isSystemic, queryLibrary, queryScope, queryCompiler);
+        return new ExecutionContext(isSystemic, authContext, queryLibrary, queryScope, queryCompiler);
+    }
+
+    /**
+     * Returns, or creates, an execution context with the given value for {@code authContext} and existing values for
+     * the other members. This is not intended to be used by user code.
+     *
+     * @param authContext the auth context to use instead
+     * @return the execution context
+     */
+    public ExecutionContext withAuthContext(final AuthContext authContext) {
+        // TODO: should the result be a composition of existing auth context and the provided?
```

Comment:
I don't think that decision belongs at this level. If we need composition, the caller should be responsible IMO:

```java
context.withAuthContext(combined(existing, newAuthContext))
```